lib/src/transformer/declaration_parsing.dart

```diff
@@ -53,11 +53,20 @@ class ParsedDeclarations {
       key_stateMixin:        <CompilationUnitMember>[],
     };
 
+    var privatePropsAndStateClassPattern = new RegExp(r'^.*class\s(_[$].*?\s)');
+
     unit.declarations.forEach((CompilationUnitMember member) {
       member.metadata.forEach((annotation) {
         var name = annotation.name.toString();
+        var match = privatePropsAndStateClassPattern.firstMatch(member.toString());
+
+        if ((name == 'Props' || name == 'State') && match != null) {
+          var publicMember = unit.declarations.where((CompilationUnitMember member) => member.toString().contains(match.group(1).substring(2)));
```

Comment:
I think this part could be broken up a bit to make the intention a bit more clear:

```dart
var publicName = match.group(1).replaceFirst('_$', '');
var publicMember = unit.declarations.firstWhere((CompilationUnitMember member) => member is ClassOrMixinDeclaration && member.name.name == publicName), orElse: () => null);
declarationMap[name]?.add(publicMember ?? member);
```